plugins/banners/create.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/banners',
+    config: {
+        description: 'Create a new banner',
+        notes: 'Create a specific banner',
+        tags: ['api', 'banners'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const { bannerFactory } = request.server.app;
+            const username = request.auth.credentials.username;
+            const scmContext = request.auth.credentials.scmContext;
+
+            // lookup whether user is admin
+            const adminDetails = request.server.plugins.banners
+                .screwdriverAdminDetails(username, scmContext);
+
+            // verify user is authorized to create banners
+            // return unauthorized if not system admin
+            if (!adminDetails.isAdmin) {
+                return reply(boom.forbidden(
+                    `User ${adminDetails.userDisplayName} is not allowed access`
```

Comment:
Wondering if it would be better to have the error message be more specific, like `User ${adminDetails.userDisplayName} does not have Screwdriver admin permissions`
Not sure if that's better or not